lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -244,6 +244,13 @@ SmallVector<unsigned> getWarpOrder(Attribute layout) {
       order.erase(it);
       order.insert(order.begin(), 0);
     }
+  } else if (auto dotOpLayout = dyn_cast<DotOperandEncodingAttr>(layout)) {
+    // opIdx=0: [batch, m, k] -> [2, 0, 1]
+    // opIdx=1: [batch, k, n] -> [2, 1, 0]
+    std::iota(order.rbegin(), order.rend(), 0);
+    if (dotOpLayout.getOpIdx() == 0) {
+      std::swap(order[0], order[1]);
+    }
```

Comment:
This will result in 
opIdx=0: [1, 2, 0]
opIdx=1: [2, 1, 0]

And I assume batch is dim0, so maybe you want the following?
opIdx=0: [1, 0, 2]
opIdx=1: [0, 1, 2]

Reply:
Yep. What I was trying to say is order[rank - 2] and order[rank - 1]. Thanks for the review.

Reply:
Because this is swapping the `order` array, I think the above code is correct.
`order[0]` refers to the leading dimension.
For opIdx=0, `[/*dim0*/batch, /*dim1=*/m, /*dim2=*/k]`, the leading dimension should be dim1=m
For opIdx=1, `[/*dim0*/batch, /*dim1=*/m, /*dim2=*/k]`, the leading dimension should be dim2=n